lib/js/src/rpc/enums/DisplayType.js

```diff
@@ -0,0 +1,148 @@
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { Enum } from '../../util/Enum.js';
+
+/**
+ * @typedef {Enum} DisplayType
+ * @property {Object} MAP
+ */
+class DisplayType extends Enum {
+
+    constructor() {
+        super();
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get CID() {
+        return DisplayType.MAP.CID;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get TYPE2() {
+        return DisplayType.MAP.TYPE2;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get TYPE5() {
+        return DisplayType.MAP.TYPE5;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get NGN() {
+        return DisplayType.MAP.NGN;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get GEN2_8_DMA() {
+        return DisplayType.MAP.GEN2_8_DMA;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get GEN2_6_DMA() {
+        return DisplayType.MAP.GEN2_6_DMA;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get MFD3() {
+        return DisplayType.MAP.MFD3;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get MFD4() {
+        return DisplayType.MAP.MFD4;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get MFD5() {
+        return DisplayType.MAP.MFD5;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get GEN3_8_INCH() {
+        return DisplayType.MAP.GEN3_8 - INCH;
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get SDL_GENERIC() {
+        return DisplayType.MAP.SDL_GENERIC;
+    }
+
+
+    /**
+    * Confirms whether the value passed in exists in the Enums of this class
+    * @param {String} value
+    * @return {null|String} - Returns null if the enum value doesn't exist
+    */
+    static valueForString(value) {
+        return DisplayType.valueForStringInternal(value, DisplayType.MAP);
+    }
+}
+
+DisplayType.MAP = Object.freeze({
+    'CID': 'CID',
+    'TYPE2': 'TYPE2',
+    'TYPE5': 'TYPE5',
+    'NGN': 'NGN',
+    'GEN2_8_DMA': 'GEN2_8_DMA',
+    'GEN2_6_DMA': 'GEN2_6_DMA',
+    'MFD3': 'MFD3',
+    'MFD4': 'MFD4',
+    'TESTING': 'TESTING',
+    'MFD5': 'MFD5',
+    'GEN3_8-INCH': 'GEN3_8_INCH',
```

Comment:
This was referencing the RPC spec. I had the values switched but if you look at the RPC spec you will see that some of these items have hyphens vs underscores and other strange things.

Reply:
```
    <enum name="DisplayType" deprecated="true" since="5.0">
        <description>See DAES for further infos regarding the displays</description>
        <element name="CID"/>
        <element name="TYPE2" />
        <element name="TYPE5" />
        <element name="NGN" />
        <element name="GEN2_8_DMA" since="3.0" />
        <element name="GEN2_6_DMA" since="3.0" />
        <element name="MFD3" since="2.0" />
        <element name="MFD4" since="2.0" />
        <element name="MFD5" since="2.0" />
        <element name="GEN3_8-INCH" internal_name="GEN3_8_INCH" since="3.0" />
        <element name="SDL_GENERIC" since="4.0" />
        <history>
            <enum name="DisplayType" since="1.0" until="5.0"/>
        </history>
    </enum>
```
I suppose `internal_name` as a key should fix the problem, `name` should be a value, not the, a key. Also `internal_name` ideally fits the naming of getter/setter methods

Reply:
Yes, it looks like the `internal_name` is ideal for keys and getter/setter method names. Would this approach match the design patterns of the other SDKs?